src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/fetch_and_validate.tsx

```diff
@@ -19,29 +19,36 @@ import {
 } from 'rxjs';
 
 import { PublishingSubject } from '@kbn/presentation-publishing';
+import {
+  OptionsListSearchTechnique,
+  OptionsListSortingType,
+} from '../../../../common/options_list';
 import { OptionsListSuccessResponse } from '../../../../common/options_list/types';
 import { isValidSearch } from '../../../../common/options_list/is_valid_search';
 import { OptionsListSelection } from '../../../../common/options_list/options_list_selections';
 import { ControlFetchContext } from '../../../control_group/control_fetch';
-import { ControlStateManager } from '../../types';
 import { OptionsListFetchCache } from './options_list_fetch_cache';
-import { OptionsListComponentApi, OptionsListComponentState, OptionsListControlApi } from './types';
+import { OptionsListComponentApi, OptionsListControlApi } from './types';
 
 export function fetchAndValidate$({
   api,
-  stateManager,
+  requestSize$,
```

Comment:
nit: why pass these in separately rather than passing in the whole API?

Reply:
I wanted to make the function dependent on the observables only and not the state manager API, which includes setters. I could update this to just pick the observables if that would be preferred.

Reply:
Is this for testing purposes? If not, having extra dependencies doesn't hurt too much IMO

Reply:
I could pass these in as state managers, but considering state is in 3 separate managers, I am not sure it would make the interface clearer. What are your thoughts?

```
       requestSize$: temporaryStateManager.api.requestSize$,
        runPastTimeout$: editorStateManager.api.runPastTimeout$,
        selectedOptions$: selectionsManager.api.selectedOptions$,
        searchTechnique$: editorStateManager.api.searchTechnique$,
```